### PR TITLE
Decoupling actions node tree

### PR DESCRIPTION
**Pull request by VictoriaShyika**

### CODE REVIEW

File: src/components/main/actionsPanel/nodeTreeView/hooks/useNodeActions.ts [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+          allow = false;
+          return;
+        }
+      });
+      if (allow) {
+        addRunningActions(["nodeTreeView-remove"]);
+
+        // call api
+        const tree = JSON.parse(JSON.stringify(nodeTree)) as TNodeTreeData;
+        const res = removeNode(tree, uids, "html");
```
[the remark this answers is not in the document]
Comment (atulbhatt-system32) [reply src/components/main/actionsPanel/nodeTreeView/hooks/useNodeActions.ts:198]: Also check that can be use native structured clone to deepclone in the line 197

File: src/components/main/actionsPanel/nodeTreeView/hooks/useNodeActions.ts [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+          nodeType,
+          htmlReferenceData,
+          osType,
+        );
+      let tempTree;
+
+      // call api
+      const tree = JSON.parse(JSON.stringify(nodeTree));
+
+      if (_tree) {
```
[the remark this answers is not in the document]
Comment (VictoriaShyika) [reply src/components/main/actionsPanel/nodeTreeView/hooks/useNodeActions.ts:73]: I also think that it should be divided, so I took this functionality to helpers

File: src/components/main/actionsPanel/nodeTreeView/hooks/useNodeActions.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (VictoriaShyika) [reply src/components/main/actionsPanel/nodeTreeView/hooks/useNodeActions.ts:135]: Also brought this functionality to helpers.
I also checked structured clone, it works well, so I decided to create a separate reusable function that gets a tree.

File: src/components/main/actionsPanel/nodeTreeView/hooks/useNodeActions.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (atulbhatt-system32) [reply src/components/main/actionsPanel/nodeTreeView/hooks/useNodeActions.ts:73]: Great

File: src/components/main/actionsPanel/nodeTreeView/hooks/useNodeActions.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (atulbhatt-system32) [reply src/components/main/actionsPanel/nodeTreeView/hooks/useNodeActions.ts:135]: good to know.

**Review by atulbhatt-system32 — APPROVED** [review]

**Comment by edenvidal** [comment]
amazing work there 🙌

**Pull request merged**